### PR TITLE
Custom Stereo nodelet

### PR DESCRIPTION
**Pull request by servos**
This is still very very rough but it currently get the job done.

This combines the two cameras into a single nodelet and uses software triggering to synchronize them.

@efernandez

### CODE REVIEW

File: pointgrey_camera_driver/nodelet_plugins.xml
Comment (efernandez) [inline pointgrey_camera_driver/nodelet_plugins.xml]: pedantic: `This is the nodelet for the Point Grey Custom Stereo Camera Driver.`

File: pointgrey_camera_driver/src/custom_stereo.cpp
Comment (efernandez) [inline pointgrey_camera_driver/src/custom_stereo.cpp:523]: :+1: For using the logic from the monocular cameras, that adds recovery when the cameras fail (stereo didn't have that).

:+1: For adding `publishImage`, we should do the same in the monocular camera (not in this PR)

File: pointgrey_camera_driver/src/custom_stereo.cpp
Comment (efernandez) [inline pointgrey_camera_driver/src/custom_stereo.cpp:512]: optional: Maybe split in two, so you can mentioned in the error message which camera (left or right) couldn't be triggered.

File: pointgrey_camera_driver/src/custom_stereo.cpp
Comment (efernandez) [inline pointgrey_camera_driver/src/custom_stereo.cpp:517]: Does it make sense to continue when one of the cameras isn't triggered?

Maybe just do nothing, or go into a recovery state. You know better how the triggering behaves, so I leave this to you.

**Review by efernandez — APPROVED** [review]

**Comment by efernandez** [comment]
Looks good to me.

Let me know if you'd do any of the minor comments I mentioned or not. Then I'd merge it, or just merge it yourself if you have merge/write access to the repo.

@mikepurvis , do you want to add something?

File: pointgrey_camera_driver/src/custom_stereo.cpp
Comment (mikepurvis) [inline pointgrey_camera_driver/src/custom_stereo.cpp]: Please update the date and add yourself as an author (Chad should probably remain given that this is clearly derived work).

File: pointgrey_camera_driver/src/custom_stereo.cpp
Comment (mikepurvis) [inline pointgrey_camera_driver/src/custom_stereo.cpp]: Remove this if you're not using it.

**Comment by servos** [comment]
I am going to decline this PR because it turned out the software sync doesn't actually work that well and this is a bit of a hack job as it is.

**Comment by efernandez** [comment]
OK, but the recovery code should be added to the stereo camera. That would be another PR though, with changes into the stereo driver.

**Comment by servos** [comment]
We don't currently use the stereo driver at all so TBH it is not a high priority.

**Pull request closed without merging**
